libraries/RC_Channel/AP_RCSwitch.cpp

```diff
@@ -0,0 +1,226 @@
+    }
+}
+
+
+void AP_RCSwitch::init()
+{
+    for (uint8_t i=0; i<NUM_RC_CHANNELS; i++) {
+        const RC_Channel *channel = AP::rc().rc_channel(i);
+        if (channel == nullptr) {
+            continue;
+        }
+        const aux_switch_pos_t position = read_3pos_switch(channel);
+        set_old_switch_position(i, position);
+        init_aux_function((aux_func_t)channel->option.get(), position);
+   }
+   reset_mode_switch();
+}
+
+// read_3pos_switch
+AP_RCSwitch::aux_switch_pos_t AP_RCSwitch::read_3pos_switch(const RC_Channel *channel) const
+{
+    const uint16_t in = channel->get_radio_in();
+    if (in < AUX_PWM_TRIGGER_LOW) return LOW;   // switch is in low position
+    if (in > AUX_PWM_TRIGGER_HIGH) return HIGH; // switch is in high position
+    return MIDDLE;                              // switch is in middle position
+}
+
+RC_Channel *AP_RCSwitch::find_channel_for_option(const aux_func_t option) const
+{
+    for (uint8_t i=0; i<NUM_RC_CHANNELS; i++) {
+        RC_Channel *channel = AP::rc().rc_channel(i);
+        if (channel == nullptr) {
+            // odd?
+            continue;
+        }
+        if ((aux_func_t)channel->option.get() == option) {
+            return channel;
+        }
+    }
+    return nullptr;
+}
+
+// duplicate_options_exist - returns true if any options are duplicated
+bool AP_RCSwitch::duplicate_options_exist() const
+{
+    uint8_t auxsw_option_counts[256] = {};
+    for (uint8_t i=0; i<NUM_RC_CHANNELS; i++) {
+        RC_Channel *channel = AP::rc().rc_channel(i);
+        if (channel == nullptr) {
+            // odd?
+            continue;
+        }
+        uint16_t option = channel->option.get();
+        if (option > sizeof(auxsw_option_counts)) {
+            AP_HAL::panic("8-bit limit on options");
+        }
+        auxsw_option_counts[option]++;
+    }
+
+    for (uint16_t i=0; i<sizeof(auxsw_option_counts); i++) {
+        if (i == 0) { // MAGIC VALUE! This is AUXSW_DO_NOTHING
+            continue;
+        }
+        if (auxsw_option_counts[i] > 1) {
+            return true;
+        }
+    }
+   return false;
+}
+
+RC_Channel *AP_RCSwitch::flight_mode_channel()
+{
+    int8_t num = flight_mode_channel_number();
+    if (num <= 0) {
+        return nullptr;
+    }
+    if (num >= NUM_RC_CHANNELS) {
+        return nullptr;
+    }
+    return AP::rc().rc_channel(num-1);
+}
+
+void AP_RCSwitch::reset_mode_switch()
+{
+    mode_switch_state.last_position = -1;
+    mode_switch_state.debounced_position = -1;
+    read_mode_switch();
+}
+
+void AP_RCSwitch::read_mode_switch()
+{
+    const RC_Channel *channel = flight_mode_channel();
+    if (channel == nullptr) {
+        return;
+    }
+
+    // calculate position of flight mode switch
+    const uint16_t pulsewidth = channel->get_radio_in();
+    if (pulsewidth <= 900 || pulsewidth >= 2200) {
+        return;  // This is an error condition
+    }
+
+    modeswitch_pos_t position;
+    if      (pulsewidth < 1231) position = 0;
```

Comment:
miss {}

Reply:
this is an exception where I think it's OK to not have the curly brackets.  It's easier to understand the way it's written here (which is the same as it is in master at the moment).